deformation_measurement/DIC_NR_images.py

```diff
@@ -200,24 +259,37 @@ def calculate(self):
 
 				#_______STORE RESULTS AND PREPARE INDICES OF NEXT SUBSET__________________
 				# Store the current displacements
-				DEFORMATION_PARAMETERS[yy,xx,0]  = self.q_k[0] # displacement x
-				DEFORMATION_PARAMETERS[yy,xx,1]  = self.q_k[1] # displacement y
-				DEFORMATION_PARAMETERS[yy,xx,2]  = self.q_k[2] 
-				DEFORMATION_PARAMETERS[yy,xx,3]  = self.q_k[3] 
-				DEFORMATION_PARAMETERS[yy,xx,4]  = self.q_k[4] 
-				DEFORMATION_PARAMETERS[yy,xx,5]  = self.q_k[5] 
-				DEFORMATION_PARAMETERS[yy,xx,6]  = 1 - C # correlation co-efficient final value
+				self.DEFORMATION_PARAMETERS[yy,xx,0]  = self.q_k[0] # displacement x
+				self.DEFORMATION_PARAMETERS[yy,xx,1]  = self.q_k[1] # displacement y
+				self.DEFORMATION_PARAMETERS[yy,xx,2]  = self.q_k[2] 
+				self.DEFORMATION_PARAMETERS[yy,xx,3]  = self.q_k[3] 
+				self.DEFORMATION_PARAMETERS[yy,xx,4]  = self.q_k[4] 
+				self.DEFORMATION_PARAMETERS[yy,xx,5]  = self.q_k[5] 
+				self.DEFORMATION_PARAMETERS[yy,xx,6]  = 1 - C # correlation co-efficient final value
 
 				# store points which are correlated in reference image i.e. center of subset
-				DEFORMATION_PARAMETERS[yy,xx,7]  = self.Xp
-				DEFORMATION_PARAMETERS[yy,xx,8]  = self.Yp
+				self.DEFORMATION_PARAMETERS[yy,xx,7]  = self.Xp
+				self.DEFORMATION_PARAMETERS[yy,xx,8]  = self.Yp
 
-				DEFORMATION_PARAMETERS[yy,xx,9]  = n # number of iterations
-				DEFORMATION_PARAMETERS[yy,xx,10] = start.total_seconds() #t_tmp # time of spline process
-				DEFORMATION_PARAMETERS[yy,xx,11] = end.total_seconds() #t_optim #time of optimization process
+				self.DEFORMATION_PARAMETERS[yy,xx,9]  = n # number of iterations
+				self.DEFORMATION_PARAMETERS[yy,xx,10] = start.total_seconds() #t_tmp # time of spline process
+				self.DEFORMATION_PARAMETERS[yy,xx,11] = end.total_seconds() #t_optim #time of optimization process
 
 			if self.debug:
 				print(yy)
 				print(xx)
 
-		return DEFORMATION_PARAMETERS
\ No newline at end of file
+	def calculate(self):
+		if not self.initialised:
+			raise error("Tried to run calculate before setting parameters. Please use set_parameters first.")
+
+		self.DEFORMATION_PARAMETERS = np.zeros((self.Y_size,self.X_size,12), dtype = float)
+
+		calc_start_time = datetime.now()
+		if (self.parallel == False):
+			self.sequential_calculate(calc_start_time)
+		else:
+			num_cores = multiprocessing.cpu_count()
+			self.DEFORMATION_PARAMETERS = Parallel(n_jobs=num_cores)(delayed(self.parallel_calculate_helper)(i, j, calc_start_time) for i in range(self.Xmin, self.Xmax) for j in range(self.Ymin,self.Ymax))
```

Comment:
I'm not super keen on having this all on one line

Reply:
Its unfortunately the only way...

Reply:
List comprehension is the only way it parallelizes from my limited research

Reply:
Ah well as long as it works haha